x-pack/legacy/plugins/siem/public/components/ml/api/throw_if_not_ok.ts

```diff
@@ -68,33 +68,31 @@ export const tryParseResponse = (response: string): string => {
 };
 
 export const throwIfErrorAttachedToSetup = (setupResponse: SetupMlResponse): void => {
-  const jobErrors = setupResponse.jobs.reduce<string[]>((accum, job) => {
-    if (job.error != null) {
-      accum = [
-        ...accum,
-        job.error.msg,
-        tryParseResponse(job.error.response),
-        `${i18n.STATUS_CODE} ${job.error.statusCode}`,
-      ];
-      return accum;
-    } else {
-      return accum;
-    }
-  }, []);
+  const jobErrors = setupResponse.jobs.reduce<string[]>(
+    (accum, job) =>
+      job.error != null
+        ? [
+            ...accum,
+            job.error.msg,
+            tryParseResponse(job.error.response),
+            `${i18n.STATUS_CODE} ${job.error.statusCode}`,
+          ]
+        : accum,
+    []
+  );
 
-  const dataFeedErrors = setupResponse.datafeeds.reduce<string[]>((accum, dataFeed) => {
-    if (dataFeed.error != null) {
-      accum = [
-        ...accum,
-        dataFeed.error.msg,
-        tryParseResponse(dataFeed.error.response),
-        `${i18n.STATUS_CODE} ${dataFeed.error.statusCode}`,
-      ];
-      return accum;
-    } else {
-      return accum;
-    }
-  }, []);
+  const dataFeedErrors = setupResponse.datafeeds.reduce<string[]>(
+    (accum, dataFeed) =>
+      dataFeed.error != null
+        ? [
+            ...accum,
+            dataFeed.error.msg,
+            tryParseResponse(dataFeed.error.response),
+            `${i18n.STATUS_CODE} ${dataFeed.error.statusCode}`,
+          ]
+        : accum,
+    []
+  );
```

Comment:
That actually looks cleaner to me, less characters, less state, and more straight forward once you understand that it's an implicit return.

Reply:
Thanks :)